am-lib/src/main/java/uk/gov/hmcts/reform/amlib/repositories/AccessManagementRepository.java

```diff
@@ -45,4 +45,7 @@
     @SqlQuery("select * from default_permissions_for_roles where service_name = :serviceName and resource_type = :resourceType and resource_name = :resourceName and role_name = :roleName")
     @RegisterConstructorMapper(RoleBasedAccessRecord.class)
     List<RoleBasedAccessRecord> getRolePermissions(String serviceName, String resourceType, String resourceName, String roleName);
+
+    @SqlQuery("select access_management_type from roles where role_name = :roleName")
+    String getRoleAccessType(String roleName);
```

Comment:
Please return enum type. Using both raw and right types brings a mess unnecessarily. Custom mapper should be the right solution. I will require us to use these for other enums as well.